test/test-runtime-onMessage.js

```diff
@@ -0,0 +1,108 @@
+"use strict";
+
+const {assert} = require("chai");
+const sinon = require("sinon");
+
+const {setupTestDOMWindow} = require("./setup");
+
+describe("browser-polyfill", () => {
+  describe("wrapped runtime.onMessage listener", () => {
+    it("keep track of the listeners added", () => {
+      const messageListener = sinon.spy();
+
+      const fakeChrome = {
+        runtime: {
+          lastError: undefined,
+          onMessage: {
+            addListener: sinon.spy(),
+            hasListener: sinon.stub(),
+            removeListener: sinon.spy(),
+          },
+        },
+      };
+
+      return setupTestDOMWindow(fakeChrome).then(window => {
+        fakeChrome.runtime.onMessage.hasListener
+          .onFirstCall().returns(false)
+          .onSecondCall().returns(true);
```

Comment:
the hasListener is a sinon stub, and so it doesn't really track the added listeners:
the first call returns false because the listener has not been added yet, the second call returns true because the listener has been added.

I added some inline comment on this stub definition, so that it is more clear the behavior that we want from this stub (and I added a third call which returns false, to cover the third call received after the listener has been removed)